thread/std-compat.h

```diff
@@ -349,6 +350,66 @@ int work_pool_init(int vcpu_num = 1, int event_engine = photon::INIT_EVENT_DEFAU
  */
 int work_pool_fini();
 
+template<typename T>
+class future {
+    using pf = photon::Future<T>;
+    std::shared_ptr<pf> _fut;
+public:
+    future(std::shared_ptr<pf>& fut) : _fut(fut) { }
+    bool valid() { return true; }
+    T get() { return _fut->get(); }
+
+    std::future_status wait(uint64_t timeout = -1) const {
+        int ret = _fut->wait(timeout);
+        return (ret == 0) ? std::future_status::ready  :
+                            std::future_status::timeout;
+    }
+
+    template< class Rep, class Period >
+    std::future_status wait_for( const std::chrono::duration<Rep,Period>& duration ) const {
+        return wait(__duration_to_microseconds(duration));
+    }
+
+    template< class Clock, class Duration >
+    std::future_status wait_until( const std::chrono::time_point<Clock,Duration>& timeout_time ) const {
+        return wait_for(timeout_time - std::chrono::system_clock::now());
```

Comment:
once timeout_time is before current time point, might should not wait.

Since `timeout_time` takes `Clock` template parameter, it should compare to `Clock::now` as well, instead of `system_clock`.